### ISSUE TITLE
[Documentation] Document how to access the config.log

### ISSUE THREAD

**Issue by ShalokShalom** [habitat-sh/habitat#2584]
Currently, the error message "look into the config.log" leaves you without any hint, where this file is and how to open it. Maybe add a prompt which directly asks the user to do that, like:

Do you want to open the config.log? (Export as text file to ..?)

**Comment by eeyun** [comment]
@ShalokShalom can you update this issue with some context on where you're seeing this error? In the meantime i'll search for that output but I want to make sure this gets triaged into the apropriate bucket!
Thanks!

**Comment by ShalokShalom** [comment]
I think i can show it to you the next time i run into it. 

The error output was "See `config.log' for more details on an error message"

And ssd means: 

If you are talking about the output of running ./configure during some build, that will usually be in the src dir in the hab cache

/hab/cache/src/DIR_FOR_YOUR_SOFTWARE/ 

I think that each and every possible case, where a text document contains information, should get covered by the documentation. I think a prompt like: Do you want to read the config.log Y/N is nice. :)

**Comment by nellshamrell** [comment]
This is not directly a result of Habitat, it is related to ./configure (autoconf and automake tools), it may be good to link to external documentation around these tools.

**Comment by ShalokShalom** [comment]
When you are in Habitat, is the usage of an editor etc different as in a casual shell, right?
And this config.log is placed in Habitats studio chroot, yes? So what is the real issue, to write something simple like a short documentation page, which describes how to access this log in Habitats special case?

What is the challenge to ask the user with the help of a simple Y/N prompt question, if he/she likes to read this log? Can we export this log maybe (on question) into the home directory of the user?

**Comment by eeyun** [comment]
@ShalokShalom would it be good to have a blog-post/series on building from these toolchains inside habitat?

To answer your second paragraph. It's not so simple as to provide a Y/N prompt question, it would require some way to run ALL commands inside of studio builds with expects and since this is a single toolchain of which habitat supports many we likely cannot safely make an assumption like that. The log location for this is set by the toolchain generating the log, not hab so in that regard it again comes down to needing to have an idea of the toolchain and the software you're building and not explicitly habitat.

**Comment by ShalokShalom** [comment]
Ok, well explained, thanks a lot

If there is no other way to offer this as an optional choice, i think we can close this. 

What do you think in general, about forks who implement specific error and solution outputs, like that one i described? As an example: Find the config log here: /path/whatsover/package_name and so on? :)

**Comment by ShalokShalom** [comment]
About the toolchains: IMHO is Habitat with that current solution some kind like a caveman with a stealth jacket. I know, most of you core maintainers like it to be so (whyever) and i still think, this is first of not compatible with the very own goals of Habitat itself: 

![2017-06-29 13_51_26-greenshot](https://user-images.githubusercontent.com/6344099/27686239-31410f5c-5cd2-11e7-8783-257886b90388.png)

And secondly, very outdated at all. Even complete outdated package managers like "place in what ever you like" offer support for all this stuff and i am very fine with a package manager, which abstracts the autotool stuff away from me.

As already said in another issue: This here IS ALL ABOUT abstraction. This whole project goes into a higher level of deployment. So why refuse user ideas with such a generic "get rid of it" answer?

It might be, that i dont understand the difference between abstraction and abstraction yet, if so, i welcome any clearance.

**Comment by eeyun** [comment]
I don't think we need to close the issue here because I think we SHOULD document how to access this log. I'm just not sure implementation wise how we could tighten control over those behaviors/errors. That is to say if someone on the team or in the community writes a PR that makes sense I don't imagine there'd be too many concerns about merging a feature like that.

**Comment by ShalokShalom** [comment]
While i welcome this very much, since i still think it makes a lot of sense, changed my view on this one a bit now: 

Together with this log, comes another message. namely the ./configure issue in this case. 

If you ask me, is this the real issue. 

I opened that issue here 16 days ago, now in the meanwhile i am aware of the reason for its fail. The nub of matter is for me: Neither the  ./configure related error message, neither the config.log can provide me any senseful info about what i can do to solve it. Without the community, i would be lost. 

Here are 2 issues with that: The community wont help again and again with the same basic questions. 
This is why we implement documentation. The second one is of course, that the very most people are too ashamed to ask in the chat, or they simply dont even this about the possibility in this moment. 

A google search will show up less. 

A simple "add ... to your pkg_build_deps" would have worked perfectly.
Why is it such hard, to implement this? All the reasons together, for why this can fail, listed in the most likely order and offer solutions for it. 

> This here is a list of potential solutions:
> 1) Check your pkg_build_deps "core/make" is an essential package for the most cases.
> 2) Check if the file name is exactly the same as in pkg_name (Case sensitivity)
> 3) Check if autogen.sh can help (Link to the specific wiki page)

I know this is something very obvious for you and everyone else in this business.  
For us, as beginners, is it very helpful. 

P.S: At least the both first cases would be redundant with a imho sane abstraction, which means such issues drive in circles. I already hit a couple of them in a row, all connected with each other
